SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AppCenter.Utils;
+using System;
+
+namespace Microsoft.AppCenter.Windows.Shared.Utils
+{
+    /**
+     * Utility to store and retrieve values for user identifiers.
+    */
+    public partial class UserIdContext
+    {
+        // The lock is static. Instance methods are not necessarily thread safe, but static methods are
+        private static readonly object UserIdLock = new object();
+        private string _userId;
```

Comment:
No need for a backing variable, we are doing nothing in get and set so no need for lock there.

Reply:
Done